### PR TITLE
Standardise flag definitions in CLI commands

### PR DESCRIPTION
**Pull request by JoshVanL**
A general house cleaning PR that helps to break up the overloading of the `/pkg/runtime` package. See also https://github.com/dapr/dapr/pull/6323 https://github.com/dapr/dapr/issues/1172.

PR moves all flag option definitions into dedicated `/options` packages for each respective CLIs. For Daprd, this results in removing `cli.go` completely, and creating a `Config` and `internalConfig` type, with `config.go` handling defaulting.

### CODE REVIEW

File: cmd/daprd/main.go [hunk 1]
```diff
@@ -14,11 +14,16 @@ limitations under the License.
...
 
 import (
+	"fmt"
+	"os"
+
 	"go.uber.org/automaxprocs/maxprocs"
 
 	// Register all components
 	_ "github.com/dapr/dapr/cmd/daprd/components"
+	"github.com/dapr/dapr/cmd/daprd/options"
```
Comment (ItalyPaleAle) [inline cmd/daprd/main.go:24]: Nit: Please move this one line below, so to leave a newline after the import of `components`

File: pkg/runtime/config.go [hunk 2]
```diff
@@ -72,118 +89,389 @@ func (p Protocol) IsHTTP() bool {
...
+	case modes.StandaloneMode:
+		if len(intc.standalone.ResourcesPath) > 0 {
+			resiliencyConfigs := resiliencyConfig.LoadLocalResiliency(log, intc.id, intc.standalone.ResourcesPath...)
+			log.Debugf("Found %d resiliency configurations in resources path", len(resiliencyConfigs))
+			resiliencyProvider = resiliencyConfig.FromConfigurations(log, resiliencyConfigs...)
+		} else {
+			resiliencyProvider = resiliencyConfig.FromConfigurations(log)
+		}
+	}
+	log.Info("Resiliency configuration loaded")
```
Comment (ItalyPaleAle) [inline pkg/runtime/config.go:271]: Do we need this? Also because it's shown even if there's no resiliency loaded

File: pkg/runtime/runtime.go [hunk 3]
```diff
@@ -1031,10 +1031,10 @@ func (a *DaprRuntime) initProxy() {
...
 		ConnectionFactory:  a.grpc.GetGRPCConnection,
-		AppID:              a.runtimeConfig.ID,
+		AppID:              a.runtimeConfig.id,
 		ACL:                a.accessControlList,
 		Resiliency:         a.resiliency,
-		MaxRequestBodySize: a.runtimeConfig.MaxRequestBodySize,
+		MaxRequestBodySize: a.runtimeConfig.maxRequestBodySize,
 	})
 
 	log.Info("gRPC proxy enabled")
```
Comment (ItalyPaleAle) [inline pkg/runtime/runtime.go:1040]: I don't think we need this either at this point (in the past, gRPC proxying was a feature gated behind a flag, but now it's not)

**Review by ItalyPaleAle — COMMENTED** [review]
LGTM, just a few nits

**Comment by JoshVanL** [comment]
Thanks @ItalyPaleAle, PTAL

**Review by ItalyPaleAle — APPROVED** [review]

**Comment by daixiang0** [comment]
something cross with https://github.com/dapr/dapr/pull/6372

**Comment by daixiang0** [comment]
Please fix DCO and lint.

**Review by artursouza — APPROVED** [review]

**Pull request merged**